File: app/src/main/java/sugtao4423/lod/swipe_image_viewer/ImagePagerAdapter.java
package sugtao4423.lod.swipe_image_viewer;

import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;

public class ImagePagerAdapter extends FragmentPagerAdapter{

    public static final String BUNDLE_KEY_URL = "url";

    private String[] urls;
    private ImageFragment[] fragments;

    public ImagePagerAdapter(FragmentManager fm, String[] urls){
        super(fm);
        this.urls = urls;
        fragments = new ImageFragment[urls.length];
        for(int i = 0; i < urls.length; i++){
            fragments[i] = new ImageFragment();
            Bundle bundle = new Bundle();
            bundle.putString(BUNDLE_KEY_URL, urls[i]);
            fragments[i].setArguments(bundle);
        }
    }

    @Override
    public Fragment getItem(int i){
        return fragments[i];
    }

    @Override
    public int getCount(){
        return urls.length;
    }

    @Override
    public CharSequence getPageTitle(int position){
        return (position + 1) + "/" + getCount();
    }

}

File: app/src/main/java/sugtao4423/lod/swipe_image_viewer/ImagePagerAdapter.kt
package sugtao4423.lod.swipe_image_viewer

import android.os.Bundle
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter

class ImagePagerAdapter(fm: FragmentManager, private val urls: Array<String>) : FragmentPagerAdapter(fm, BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT) {

    companion object {
        const val BUNDLE_KEY_URL = "url"
    }

    private val fragments: Array<ImageFragment>

    init {
        val fragments = arrayOfNulls<ImageFragment>(urls.size)
        for (i in urls.indices) {
            val bundle = Bundle().apply {
                putString(BUNDLE_KEY_URL, urls[i])
            }
            fragments[i] = ImageFragment()
            fragments[i]!!.arguments = bundle
        }
        this.fragments = fragments.requireNoNulls()
    }

    override fun getItem(i: Int): Fragment {
        return fragments[i]
    }

    override fun getCount(): Int {
        return urls.size
    }

    override fun getPageTitle(position: Int): CharSequence {
        return "${position + 1}/$count"
    }

}